tests/ui/type-alias-impl-trait/issue-109054.rs

```diff
@@ -11,6 +11,7 @@ impl std::ops::Deref for CallMe {
     type Target = FnType;
 
     fn deref(&self) -> &Self::Target {
+        //~^ ERROR: item does not constrain `ReturnType
         fn inner(val: &u32) -> ReturnType {
```

Comment:
that one is an interesting pattern 🤔 how would we support that one with a `defines` attribute?